HerPublicWebsite.UnitTests/BusinessLogic/Services/PendingReferralFilterServiceTests.cs

```diff
@@ -0,0 +1,91 @@
+
+    [SetUp]
+    public void Setup()
+    {
+        mockDateHelper = new Mock<IDateHelper>();
+        pendingReferralFilterService = new PendingReferralFilterService(mockDateHelper.Object);
+    }
+
+    [TestCase(false, true, true)] // case 2
+    [TestCase(true, false, false)] // case 1
+    [TestCase(true, false, true)] // case 1, 3
+    [TestCase(true, true, false)] // case 1
+    [TestCase(true, true, true)] // case 1
+    public void PendingReferralFilterService_WhenCalledWithEmailPendingReferral_IncludesInFilter(
+        bool localAuthorityIsNowPending,
+        bool localAuthorityWasPending,
+        bool referralWasSubmittedInTheLastMonth)
+    {
+        SetupReferralData(localAuthorityIsNowPending, localAuthorityWasPending, referralWasSubmittedInTheLastMonth);
+        var inputReferralRequests = new List<ReferralRequest>() { inputReferralRequest };
+        
+        // Act
+        var outputReferralRequests = pendingReferralFilterService
+            .FilterForPendingReferralReport(inputReferralRequests);
+        
+        // Assert
+        CollectionAssert.Contains(outputReferralRequests, inputReferralRequest);
+    }
+
+    [TestCase(false, false, false)]
+    [TestCase(false, false, true)]
+    [TestCase(false, true, false)]
+    public void PendingReferralFilterService_WhenCalledWithNotEmailPendingReferral_DoesNotIncludeInFilter(
+        bool localAuthorityIsNowPending,
+        bool localAuthorityWasPending,
+        bool referralWasSubmittedInTheLastMonth)
+    {
+        SetupReferralData(localAuthorityIsNowPending, localAuthorityWasPending, referralWasSubmittedInTheLastMonth);
+        var inputReferralRequests = new List<ReferralRequest>() { inputReferralRequest };
+        
+        // Act
+        var outputReferralRequests = pendingReferralFilterService
+            .FilterForPendingReferralReport(inputReferralRequests);
+        
+        // Assert
+        CollectionAssert.DoesNotContain(outputReferralRequests, inputReferralRequest);
+    }
+
+    private void SetupReferralData(
+        bool localAuthorityIsNowPending,
+        bool localAuthorityWasPending,
+        bool referralWasSubmittedInTheLastMonth)
+    {
+        // Arrange
+        var custodianCode =
+            LocalAuthorityDataHelper.GetExampleCustodianCodeForStatus(
+                localAuthorityIsNowPending
+                    ? LocalAuthorityData.Hug2Status.Pending
+                    : LocalAuthorityData.Hug2Status.Live);
+
+        var requestDate = new DateTime(
+            2024,
+            referralWasSubmittedInTheLastMonth ? 2 : 1, // as 'last month' is defined as 1st february
+            15);
+        inputReferralRequest = new ReferralRequestBuilder(10)
+            .WithCustodianCode(custodianCode)
+            .WithWasSubmittedToPendingLocalAuthority(localAuthorityWasPending)
+            .WithRequestDate(requestDate)
+            .Build();
+
+        var oneMonthAgoDate = new DateTime(2024, 2, 1);
```

Comment:
This should be `startOfPreviousMonth` to be more precise.

You can also ship this bit into the whole class `Setup()` method.

Overall, you can refactor as below.

(Note this also uses some FluentAssertions syntactic sugar for [more readable dates](https://fluentassertions.com/datetimespans/) in tests.)

```
private DateTime startOfPreviousMonth;

[Setup]
public void Setup()
{
    startOfPreviousMonth = 1.February(2024);
    mockDateHelper = new Mock<IDateHelper>();
    mockDateHelper.Setup(mdh => mdh.GetStartOfPreviousMonth()).Returns(startOfPreviousMonth);

    pendingReferralFilterService = new PendingReferralFilterService(mockDateHelper.Object);
}
```
```
public void PendingReferralFilterService_WhenCalledWithNotEmailPendingReferral_DoesNotIncludeInFilter(
    bool localAuthorityIsNowPending,
    bool localAuthorityWasPending,
    bool referralWasSubmittedInTheLastMonth)
{
    // Arrange
    var referralRequest = BuildReferralRequest(
        localAuthorityIsNowPending, localAuthorityWasPending, referralWasSubmittedInTheLastMonth);
    var inputReferralRequests = new List<ReferralRequest> { referralRequest };
```
```
private ReferralRequest BuildReferralRequest(
    bool localAuthorityIsNowPending,
    bool localAuthorityWasPending,
    bool referralWasSubmittedInTheLastMonth)
{
    var currentStatus = localAuthorityIsNowPending ? Hug2Status.Pending : Hug2Status.Live;
    var custodianCode = LocalAuthorityDataHelper.GetExampleCustodianCodeForStatus(currentStatus);

    var requestDate = referralWasSubmittedInTheLastMonth
        ? startOfPreviousMonth.Add(10.Days())
        : startOfPreviousMonth.Subtract(10.Days());

    return new ReferralRequestBuilder(10)
        .WithCustodianCode(custodianCode)
        .WithWasSubmittedToPendingLocalAuthority(localAuthorityWasPending)
        .WithRequestDate(requestDate)
        .Build();
}